CRM/Activity/BAO/ActivityContact.php

```diff
@@ -39,8 +39,11 @@ public function __construct() {
   public static function create($params) {
     $activityContact = new CRM_Activity_DAO_ActivityContact();
     $activityContact->copyValues($params);
+    CRM_Utils_Hook::pre('create', 'ActivityContact', NULL, $params);
```

Comment:
@yashodha won't this cover edit as well as create?

Reply:
I wonder if it's better to just do 

try {
  return self::writeRecord($params);
}
catch {

}

Reply:
@yashodha @seamuslee001 I pulled ^^ into https://github.com/civicrm/civicrm-core/pull/19623